podman/client.py

```diff
@@ -114,27 +115,38 @@ def from_env(
             Client used to communicate with a Podman service.
 
         Raises:
-            ValueError when required environment variable is not set
+            PodmanConnectionError: When connection to service fails or environment is invalid
         """
-        environment = environment or os.environ
-        credstore_env = credstore_env or {}
-
-        if version == "auto":
-            version = None
-
-        kwargs = {
-            'version': version,
-            'timeout': timeout,
-            'tls': False,
-            'credstore_env': credstore_env,
-            'max_pool_size': max_pool_size,
-        }
-
-        host = environment.get("CONTAINER_HOST") or environment.get("DOCKER_HOST") or None
-        if host is not None:
-            kwargs['base_url'] = host
-
-        return PodmanClient(**kwargs)
+        try:
+            environment = environment or os.environ
+            credstore_env = credstore_env or {}
+
+            if version == "auto":
+                version = None
+
+            kwargs = {
+                'version': version,
+                'timeout': timeout,
+                'tls': False,
+                'credstore_env': credstore_env,
+                'max_pool_size': max_pool_size,
+            }
+
+            host = environment.get("CONTAINER_HOST") or environment.get("DOCKER_HOST") or None
+            if host is not None:
+                kwargs['base_url'] = host
+
+            return PodmanClient(**kwargs)
+        except Exception as exc:
+            error_msg = "Failed to initialize Podman client from environment"
+            if isinstance(exc, ValueError):
+                error_msg = "Invalid environment configuration for Podman client"
+            elif isinstance(exc, (ConnectionError, TimeoutError)):
+                error_msg = "Failed to connect to Podman service"
+
+            raise PodmanConnectionError(
+                message=error_msg, environment=environment, host=host, original_error=exc
+            ) from exc
 
```

Comment:
For code clarity, I would prefer to use python's exception handling here.

```
except ValueError as e:
    error_msg = "Invalid environment configuration for Podman client"
    raise PodmanConnectionError(
        message=error_msg, environment=environment, host=host, original_error=e)
except (ConnectionError, TimeoutError) as e:
    error_msg = "Failed to connect to Podman service"
    raise PodmanConnectionError(
         message=error_msg, environment=environment, host=host, original_error=e)
except Exception as e:
    error_msg = "Failed to initialize Podman client from environment"
    raise PodmanConnectionError(
         message=error_msg, environment=environment, host=host, original_error=e)
```

I would avoid filtering exceptions with if conditions in the except block, unless you have a strong reason to do so.